hspec-core/src/Test/Hspec/Core/Spec/Monad.hs

```diff
@@ -66,7 +72,11 @@ runIO :: IO r -> SpecM a r
 runIO = SpecM . liftIO
 
 mapSpecForest :: ([SpecTree a] -> [SpecTree b]) -> SpecM a r -> SpecM b r
-mapSpecForest f (SpecM specs) = SpecM (mapWriterT (fmap (fmap (second f))) specs)
+mapSpecForest = mapSpecForest_ id
+
+mapSpecForest_ :: (Env -> Env) -> ([SpecTree a] -> [SpecTree b]) -> SpecM a r -> SpecM b r
+mapSpecForest_ ef tf (SpecM specs) = SpecM $ flip mapWriterT specs $ 
+  mapReaderT (fmap (fmap (second tf))) . withReaderT ef
```

Comment:
I'm not happy with this naming. Because `mapSpecItem_ = mapSpecItem id` - and the underscore variant here is flipped. I kept it as is for backwards compatibility on `mapSpecForest`, though.

Reply:
This became redundant using the `local` refactor.